File: src/main/java/me/deecaad/weaponmechanicsplus/PlusScriptManager.java
package me.deecaad.weaponmechanicsplus;

import me.deecaad.weaponmechanics.weapon.projectile.AProjectile;
import me.deecaad.weaponmechanics.weapon.projectile.ProjectileScriptManager;
import me.deecaad.weaponmechanics.weapon.projectile.weaponprojectile.WeaponProjectile;
import me.deecaad.weaponmechanicsplus.weapon.guidedprojectile.GuidedProjectileScript;
import org.bukkit.plugin.Plugin;

import javax.annotation.Nonnull;

import static me.deecaad.weaponmechanics.WeaponMechanics.getConfigurations;

public class PlusScriptManager extends ProjectileScriptManager {

    public PlusScriptManager(Plugin plugin) {
        super(plugin);
    }

    @Override
    public void attach(@Nonnull AProjectile aProjectile) {
        if (!(aProjectile instanceof WeaponProjectile projectile)) return;

        if (getConfigurations().containsKey(projectile.getWeaponTitle() + ".Projectile.Guided_Projectile")) {
            projectile.addProjectileScript(new GuidedProjectileScript(getPlugin(), projectile));
        }
    }
}

File: src/main/kotlin/me/deecaad/weaponmechanicsplus/PlusScriptManager.kt
package me.deecaad.weaponmechanicsplus

import me.deecaad.weaponmechanics.WeaponMechanics
import me.deecaad.weaponmechanics.weapon.projectile.AProjectile
import me.deecaad.weaponmechanics.weapon.projectile.ProjectileScriptManager
import me.deecaad.weaponmechanics.weapon.projectile.weaponprojectile.WeaponProjectile
import me.deecaad.weaponmechanicsplus.weapon.guidedprojectile.GuidedProjectileScript
import org.bukkit.plugin.Plugin

class PlusScriptManager(plugin: Plugin?) : ProjectileScriptManager(plugin) {

    override fun attach(aProjectile: AProjectile) {
        if (aProjectile !is WeaponProjectile)
            return

        if (WeaponMechanics.getConfigurations().containsKey(aProjectile.weaponTitle + ".Projectile.Guided_Projectile")) {
            aProjectile.addProjectileScript(GuidedProjectileScript(plugin, aProjectile))
        }
    }
}